### DISCUSSION TITLE
Running Avalonia Web App on Server

### DISCUSSION THREAD

**Discussion by LarsNijholt** [AvaloniaUI/Avalonia#10298] [Q&A]
I am using the new Avalonia 11 preview. And i am trying to get my project to run on a server(SFTP accessed by filezilla), Now it runs on my Local machine just fine. But when i copy the files from bin->release->net7.0->browser-wasm->AppBundle it does not run the project and only shows the "Powered by Avalonia" Splash screen. I have seen examples of projects running in the browser. On Github pages for example(If you know how to do this then this would also be a very welcome option/solution). But i cannot seem to figure out what i am doing wrong. Could somebody maybe explain to me how to do this?

Thanks in advance :)

**Comment by ahopper** [comment]
Do the browser dev tools give you any errors/clues?

**Reply by LarsNijholt** [reply]
This is what i get when i open the website

![AvaloniaError](https://user-images.githubusercontent.com/61054131/218135855-a31ad2db-b9f5-4f01-9256-c48babab2e93.png)

**Reply by ahopper** [reply]
you may need to set base href in index.html if you are serving it from a different path (I don't know if this still applies to recent bulds)

**Reply by MikePapinski** [reply]
I am having similar problem by trying to deploy Avalonia to firebase hosting.
When I run my build command it does not generate AppBundle at All..
When I copy files from my debug folder I got this.
![image](https://user-images.githubusercontent.com/48183086/218139405-5a529a81-1f2e-4c65-bb95-41129aa4e4a3.png)
![image](https://user-images.githubusercontent.com/48183086/218139554-53c88c50-2d8d-4628-b829-86390f61cd93.png)

**Reply by ahopper** [reply]
I assume you have installed the wasm tools - dotnet workload install wasm-tools wasm-experimental

**Reply by maxkatz6** [reply]
@LarsNijholt your error message doesn't look like anything Avalonia related. If it won't be solved, makes sense to create an issue in dotnet/runtime repository, as they maintain .NET WASM runtime support.

@MikePapinski your error message is different though. Looks like XAML code was trimmed out from the library. Are you running 11.0.0-preview5 version? It shouldn't be an issue there.

**Comment by paulgee31** [comment]
For what it is worth and it might useful to know I was able to successfully deploy the contents of the AppBundle to IIS with the following web.config file:

```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.webServer>
	 <staticContent>
                <remove fileExtension=".blat" />
                <remove fileExtension=".dat" />
                <remove fileExtension=".dll" />
                <remove fileExtension=".json" />
                <remove fileExtension=".wasm" />
                <remove fileExtension=".woff" />
                <remove fileExtension=".woff2" />
		<remove fileExtension=".ts" />
                <mimeMap fileExtension=".blat" mimeType="application/octet-stream" />
                <mimeMap fileExtension=".dll" mimeType="application/octet-stream" />
                <mimeMap fileExtension=".dat" mimeType="application/octet-stream" />
                <mimeMap fileExtension=".json" mimeType="application/json" />
                <mimeMap fileExtension=".wasm" mimeType="application/wasm" />
                <mimeMap fileExtension=".woff" mimeType="application/font-woff" />
                <mimeMap fileExtension=".woff2" mimeType="application/font-woff" />
		<mimeMap fileExtension=".ts" mimeType="application/x-typescript" />
		<mimeMap fileExtension=".symbols" mimeType="text/plain" />
	 </staticContent>
         <httpCompression>
          	 <dynamicTypes>
                    <add mimeType="application/octet-stream" enabled="true" />
                    <add mimeType="application/wasm" enabled="true" />
          	 </dynamicTypes>
          </httpCompression>
    </system.webServer>
</configuration>
```
